images/spire/main.tf

```diff
@@ -1,40 +1,40 @@
-locals {
-  components = toset(["server", "agent", "oidc-discovery-provider"])
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
 }
 
 variable "target_repository" {
   description = "The docker repo into which the image and attestations should be published."
 }
 
-module "latest" {
-  for_each          = local.components
-  source            = "../../tflib/publisher"
-  name              = basename(path.module)
-  target_repository = "${var.target_repository}-${each.key}"
-  config            = file("${path.module}/configs/latest.${each.key}.apko.yaml")
-  build-dev         = true
+locals {
+  components = toset(["server", "agent", "oidc-discovery-provider"])
```

Comment:
Can this be made into a map where you also carry your `run-as` and `command`, and then send it into the `config` via variables as explained in the previous comments?

Reply:
Not sure if I understood this part, do we have any similar examples for it?

Reply:
I don't know if we currently have anything similar. What I had in mind was something around the lines of:
```terraform
locals {
  component_data = {
    "server" = {
      run-as  = 0
      command = "server"
    },
    // other components here
  }
  components = toset(keys(local.component_data))
}
```
so you won't need conditionals in the `config` module.

Reply:
Thanks @stormqueen1990 , just tried to refactor with your amazing suggestions, feel free to take a look

Reply:
it worked I guess 🥳